test/Microsoft.AspNet.Mvc.Core.Test/ControllerTests.cs

```diff
@@ -1297,6 +1297,220 @@ public void ControllerView_UsesControllerViewEngine()
             Assert.Same(viewEngine, result.ViewEngine);
         }
 
+        [Theory]
+        [MemberData(nameof(TestabilityViewTestData))]
+        public void ControllerView_InvokedInUnitTests(object modelObject, string viewName)
```

Comment:
Consider putting this in another class, so cleanly separate the 'user unit test' style code from the 'framework author unit test' style code.

It's totally ok to break up unit tests by feature, especially for something big and important. Just keep the words 'Controller' and 'Test' in the name like: ControllerUnitTestStyleTest.

Some comments would be helpful too. Any time code isn't written in a utilitarian way, you run the risk that someone else will come by and 'clean up'.